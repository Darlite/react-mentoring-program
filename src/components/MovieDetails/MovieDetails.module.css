.movieDetails {
    display: flex;
    gap: 50px;
    width: 1200px;
    align-items: flex-start;
}

.movieInfo {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
}

.movieTitle {
    margin: 0;
    text-transform: uppercase;
    font-size: 40px;
    letter-spacing: 1px;
    font-weight: 300;
    flex-basis: 92%;
}

.rating {
    border: 1px solid #FFFFFF;
    width: 40px;
    height: 40px;
    line-height: 40px;
    padding: 5px;
    font-size: 20px;
    font-weight: 300;
    border-radius: 50%;
    text-align: center;
}

.genres {
    font-size: 16px;
    font-weight: 500;
    color: #b7b7b7;
    margin: 15px 0 30px 0;
    flex-basis: 100%;
}

.releaseDate {
    color: #F65261;
    font-size: 24px;
    font-weight: 300;
    margin-right: 40px;
}

.runtime {
    color: #F65261;
    font-size: 24px;
    font-weight: 300;
    flex-basis: 70%;
}

.overview {
    font-size: 20px;
    line-height: 24px;
    font-weight: 300;
    color: #b0b0b0;
    margin: 25px 0 0;
}

.searchButton {
    background: url("../../assets/images/searchIcon.svg") 100% 50% no-repeat;
    width: 130px;
    height: 50px;
    line-height: 24px;
    cursor: pointer;
    margin-left: auto;
    margin-top: 30px;
}

.movieDetailsImage {
    width: 323px;
}
