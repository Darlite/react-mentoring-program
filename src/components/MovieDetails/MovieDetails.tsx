import styles from './MovieDetails.module.css';
import React from "react";
import {MovieDetailsData} from "../../types/MovieDetailsData";
import placeholderImage from "../../assets/images/placeholderPoster.jpg";
import formatRuntime from "../../utils/formatRuntime";

interface MovieDetailsProps {
    movieDetails: MovieDetailsData;
    handleBackToSearch: () => void;
}

const MovieDetails: React.FC<MovieDetailsProps> = ({movieDetails, handleBackToSearch}) => {
    const {poster_path, title, release_date, genres, runtime, vote_average, overview} = movieDetails;
    return (
        <div className={styles.movieDetails}>
            <img className={styles.movieDetailsImage}
                 src={poster_path}
                 alt={title}
                 onError={(e) => {
                     e.currentTarget.src = placeholderImage;
                     e.currentTarget.onerror = null;
                 }
                }
            />
            <div className={styles.movieInfo}>
                <h1 className={styles.movieTitle}>{title}</h1>
                <span className={styles.rating}>{vote_average}</span>
                <p className={styles.genres}>
                    {genres.map(genre => (
                        <span key={genre}>{genre} </span>
                    ))}
                </p>
                <span className={styles.releaseDate}>{release_date}</span>
                <span className={styles.runtime}>{formatRuntime(runtime)}</span>
                <p className={styles.overview}>{overview}</p>
                <span className={styles.searchButton}
                      role="button"
                      onClick={handleBackToSearch}>Back to the search</span>
            </div>
        </div>
    )
}

export default MovieDetails;
