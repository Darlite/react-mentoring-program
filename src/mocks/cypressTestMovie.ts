import {GenreType} from "../constants/GenreType";

export const cypressTestMovie = {
    title: "Cypress Test Movie",
    editedMovieTitle: "Cypress Edited Test Movie",
    release_date: "2025-05-04",
    poster_path: "https://www.movieposters.com/cdn/shop/files/tron-ares_avxg7mih_480x.progressive.jpg",
    genres: [GenreType.Action, GenreType.ScienceFiction],
    runtime: "138",
    vote_average: "7.2",
    overview: "A highly sophisticated Program, Ares, is sent from the digital world into the real world on a dangerous mission.",
}
