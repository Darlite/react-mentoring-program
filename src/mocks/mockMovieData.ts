export const moviesData = {
    imageUrl: "images/Pulp_Fiction.png",
    movieName: "Pulp Fiction",
    releaseYear: 1994,
    relevantGenres: ["Crime", "Thriller"],
    duration: "2h 34m",
    rating: 8.9,
    description: "Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra",
}
