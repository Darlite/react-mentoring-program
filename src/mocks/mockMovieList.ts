import {GenreType} from "../constants/GenreType";

export const mockMovieList = [
    {
        id: 1,
        poster_path: "images/Pulp_Fiction.png",
        title: "Pulp Fiction",
        release_date: "1994-09-23",
        movieUrl: "https://www.imdb.com/title/tt0110912",
        genres: [GenreType.Crime, GenreType.Thriller],
        runtime: "2h 34m",
        vote_average: "8.9",
        overview: "Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra",
    },
    {
        id: 2,
        poster_path: "images/Bohemian_Rhapsody.png",
        title: "Bohemian Rhapsody",
        release_date: "2018-10-30",
        movieUrl: "https://www.imdb.com/title/tt1727824",
        genres: [GenreType.Musical, GenreType.Documentary],
        runtime: "2h 15m",
        vote_average: "8.5",
        overview: "With his impeccable vocal abilities, Freddie Mercury and his rock band, Queen, achieve superstardom. However, amidst his skyrocketing success, he grapples with his ego, sexuality and a fatal illness.",
    },
    {
        id: 3,
        poster_path: "images/Kill_Bill_Volume_2.png",
        title: "Kill Bill: Vol. 2",
        release_date: "2004-04-08",
        movieUrl: "https://www.imdb.com/title/tt0378194",
        genres: [GenreType.Action, GenreType.Thriller],
        runtime: "2h 17m",
        vote_average: "8.9",
        overview: "A pregnant woman, codenamed the Bride, sets out on a journey to kill her ex-boss, Bill, and targets his brother, Budd, and Elle Driver, the only two survivors of the Deadly Vipers Assassination Squad.",
    },
]