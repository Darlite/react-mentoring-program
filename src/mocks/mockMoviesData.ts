import {GenreType} from "../constants/GenreType";

export const mockMoviesData = {
    id: 680,
    poster_path: "https://www.movieposters.com/cdn/shop/files/PulpFiction.Ver1_480x.progressive.jpg?v=1741192018",
    title: "Pulp Fiction",
    release_date: "1994-09-10",
    genres: [GenreType.Crime, GenreType.Thriller],
    runtime: "167",
    vote_average: "8.3",
    overview: "Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra",
}
