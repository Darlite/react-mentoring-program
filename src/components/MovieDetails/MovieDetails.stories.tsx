import { Meta, StoryObj } from "@storybook/react";
import MovieDetails from "./MovieDetails";


const meta = {
    component: MovieDetails,
    title: 'MovieDetails',
    tags: ['autodocs'],
    //ðŸ‘‡ Our exports that end in "Data" are not stories.
    excludeStories: /.*Data$/,
    args: {},
} satisfies Meta<typeof MovieDetails>;

export default meta;

type Story = StoryObj<typeof meta>;

const moviesData = {
    imageUrl: "images/Pulp_Fiction.png",
    movieName: "Pulp Fiction",
    releaseYear: 1994,
    relevantGenres: ["Crime", "Thriller"],
    duration: "2h 34m",
    rating: 8.9,
    description: "Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra",
}

export const Default: Story = {
    args: {
        movieDetails: moviesData,
    },
};
