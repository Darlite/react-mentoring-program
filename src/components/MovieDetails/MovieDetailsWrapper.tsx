import {Outlet, useLoaderData, useNavigate} from "react-router-dom";
import {MovieDetailsData} from "../../types/MovieDetailsData";
import MovieDetails from "./MovieDetails";

export default function MovieDetailsWrapper() {
    const {movieDetails} = useLoaderData() as { movieDetails: MovieDetailsData }
    const navigate = useNavigate();

    return <div>
        <MovieDetails movieDetails={movieDetails}
                      handleBackToSearch={() => navigate(-1)}/>
        <Outlet />
    </div>
}
