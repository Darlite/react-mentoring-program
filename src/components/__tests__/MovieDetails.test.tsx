import {render} from "@testing-library/react";
import MovieDetails from "../MovieDetails/MovieDetails";
import {mockMoviesData} from "../../mocks/mockMoviesData";

describe('MovieDetails', () => {
    it("renders correctly", () => {
        const { container } = render(<MovieDetails movieDetails={mockMoviesData} />);
        expect(container).toMatchSnapshot();
    });
})
