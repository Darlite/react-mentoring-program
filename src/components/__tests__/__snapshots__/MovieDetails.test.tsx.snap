// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders correctly 1`] = `
<div>
  <div
    class="movieDetails"
  >
    <img
      alt="Pulp Fiction"
      src="images/Pulp_Fiction.png"
    />
    <div
      class="movieInfo"
    >
      <h1
        class="movieTitle"
      >
        Pulp Fiction
      </h1>
      <span
        class="rating"
      >
        8.9
      </span>
      <p
        class="genres"
      >
        <span>
          Crime
           
        </span>
        <span>
          Thriller
           
        </span>
      </p>
      <span
        class="releaseYear"
      >
        1994
      </span>
      <span
        class="duration"
      >
        2h 34m
      </span>
      <p
        class="description"
      >
        Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra
      </p>
    </div>
  </div>
</div>
`;
